### ISSUE TITLE
Nest can't resolve dependencies  of the CompaniesService (?)

### ISSUE THREAD

**Issue by EasonQwQ** [nestjsx/crud#682]
I'm quite new to Nest and CRUD,and i try to follow the doc,but i get the error  
![image](https://user-images.githubusercontent.com/22360580/111513051-5ae39400-878b-11eb-897a-d7c7c508fd01.png)

this is my  companies.module.ts just copy from doc
![image](https://user-images.githubusercontent.com/22360580/111513325-a6963d80-878b-11eb-8b8f-076119807778.png)

**Comment by EasonQwQ** [comment]
```
import { Module } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';

import { Company } from './company.entity';
import { CompaniesService } from './companies.service';
import { CompaniesController } from './companies.controller';

@Module({
  imports: [TypeOrmModule.forFeature([Company])],
  providers: [CompaniesService],
  exports: [CompaniesService],
  controllers: [CompaniesController],
})
export class CompaniesModule {}
```

**Comment by EasonQwQ** [comment]
this is my  app.module.ts 
```

```

**Comment by EasonQwQ** [comment]
import { Module } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { Connection } from 'typeorm';
import { CompaniesController } from './companies/companies.controller';
import { CompaniesService } from './companies/companies.service';
import { CompaniesModule } from './companies/companies.module';
@Module({
  imports: [TypeOrmModule.forRoot(), CompaniesModule],
  controllers: [AppController, CompaniesController],
  providers: [AppService, CompaniesService],
})
export class AppModule {
  constructor(private readonly connection: Connection) {}
}

**Comment by EasonQwQ** [comment]
this is my app.module.ts 
```
import { Module } from '@nestjs/common';
import { TypeOrmModule } from '@nestjs/typeorm';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { Connection } from 'typeorm';
import { CompaniesController } from './companies/companies.controller';
import { CompaniesService } from './companies/companies.service';
import { CompaniesModule } from './companies/companies.module';
@Module({
  imports: [TypeOrmModule.forRoot(), CompaniesModule],
  controllers: [AppController, CompaniesController],
  providers: [AppService, CompaniesService],
})
export class AppModule {
  constructor(private readonly connection: Connection) {}
}
```

**Comment by bashleigh** [comment]
What does CompanyService look like?

**Comment by EasonQwQ** [comment]
> What does CompanyService look like?

```
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { TypeOrmCrudService } from '@nestjsx/crud-typeorm';

import { Company } from './company.entity';

@Injectable()
export class CompaniesService extends TypeOrmCrudService<Company> {
  constructor(@InjectRepository(Company) repo) {
    super(repo);
  }
}
``` 
just coyp from docs

**Comment by bashleigh** [comment]
You need to remove `companiesService` and `CompaniesController` from the `AppModule`

**Comment by bennycode** [comment]
**Wow!** That was very helpful @bashleigh. 👍 I came here in 2023 because I also ran into this error:

> Error: Nest can't resolve dependencies of the CategoryService (?). Please make sure that the argument CategoryRepository at index [0] is available in the AppModule context.
> 
> Potential solutions:
> - Is AppModule a valid NestJS module?
> - If CategoryRepository is a provider, is it part of the current AppModule?
> - If CategoryRepository is exported from a separate @Module, is that module imported within AppModule?

The `CategoryRepository` comes into play from the `@InjectRepository(Category)` decoration in my `CategoryService`. I thought that it must be available as I am exporting the `CategoryService` in my `CategoryModule`. The `CategoryModule` was imported in my `AppModule` but I didn't think of removing `CategoryController` and `CategoryService` from the `controllers` and `providers` list as these entries have been made when running `nest g co` and `nest g s` (as per [CRUD generation docs](https://docs.nestjs.com/recipes/crud-generator)).

Your comment helped me to solve the problem. I will check if this hint can be added to the "Potential solutions" of [Nest](https://github.com/nestjs/nest). ✍️
